userbenchmark/utils.py

```diff
@@ -85,27 +85,32 @@ def get_output_dir(bm_name) -> Path:
     return target_dir
 
 
-def get_latest_n_jsons_from_s3(n: int, bm_name: str, platform_name: str, date: str) -> List[str]:
-    """Retrieves the most recent n metrics json filenames from S3 the WEEK BEFORE the given date, exclusive of that date.
-       If fewer than n items are found, returns all found items without erroring, even if there were no items. """
+def get_date_from_metrics_s3_key(metrics_s3_key: str) -> datetime:
+    metrics_s3_json_filename = metrics_s3_key.split('/')[-1]
+    return datetime.strptime(metrics_s3_json_filename, 'metrics-%Y%m%d%H%M%S.json')
+
+
+def get_latest_n_day_jsons_from_s3(ndays: int, bm_name: str, platform_name: str, date: datetime, limit: int=100) -> List[str]:
```

Comment:
```suggestion
def get_latest_jsons_in_s3_from_last_n_days(ndays: int, bm_name: str, platform_name: str, date: datetime, limit: int=100) -> List[str]:
```
would be a clearer name